src/sugar.jl

```diff
@@ -1159,3 +1159,75 @@ grad
     )
     return nothing
 end
+
+"""
+    autodiff(
+        rmode::Union{ReverseMode,ReverseModeSplit},
+        f::Annotation,
+        dresult::Seed,
+        annotated_args...
+    )
+    
+Call [`autodiff_thunk`](@ref) in split mode, execute the forward pass, increment output adjoint with `dresult`, then execute the reverse pass.
+
+Useful for computing pullbacks / VJPs for functions whose output is not a scalar, or when the scalar seed is not 1.
+"""
+function autodiff(
+        rmode::Union{ReverseMode{ReturnPrimal}, ReverseModeSplit{ReturnPrimal}},
+        f::FA,
+        dresult::Seed{RT},
+        args::Vararg{Annotation, N},
+    ) where {ReturnPrimal, FA <: Annotation, RT, N}
+    rmode_split = Split(rmode)
+    RA = guess_activity(RT, rmode_split)
+    forward, reverse = autodiff_thunk(rmode_split, FA, RA, typeof.(args)...)
+    tape, result, shadow_result = forward(f, args...)
+    if RA <: Active
```

Comment:
If RA is active, instead of calling autodiff thunk, we can actually just call thunk and instead pass this value instead of Compiler.default_adjoint(RT):

https://github.com/EnzymeAD/Enzyme.jl/blob/7134994c921578442f3facb7b361b98878c441f5/src/Enzyme.jl#L501

That makes me think it's worthwhile to split out that inner code to a helper method that can be used here and there (or alternatively have the other autodiff actually call this method)

Reply:
> or alternatively have the other autodiff actually call this method

I think that would be good but it might be a big change, so let's maybe do it in a separate PR?

Reply:
> we can actually just call thunk and instead pass this value

I'm not sure I understand what value you're referring to

Reply:
> we can actually just call thunk and instead pass this value

The provided seed value. E.g. for `Active` Enzyme currently synthesizes a seed value of `one` and so you wouldn't need to switch to `autodiff_thunk` but could use the slightly faster combined mode

Reply:
Okay so if `autodiff_thunk` is slower then we don't want other methods to rely on my new addition, right?

Reply:
I think it's a "theoretically" slower. `CombinedAugmentedReverse` can sometimes avoid the tape and thus directly  forward values from the augmented primal to the reverse pass. 

Furthermore it can avoid caching since it can assume that data was not modified in between the augmented forward and reverse.

Reply:
So what do we do here? Do we change anything?

Reply:
okay I thought about this a bit more and the current design won't work for the proper thunk implementation

In particular I think we need Seed{Active} (e.g. for floats), Seed{Duplicated} (e.g. for vectors), Seed{BatchDuplicated} etc. These all will just contain the one object, but the type will convey the semantic meaning.

We then should have Seed(x) call guess_activity and dispatch to the right constructor

Reply:
The problem is that `guess_activity` needs a `mode` object, which is not available until we call `autodiff`. Unless we do `Seed(dy, mode)`.

Reply:
seed only applies in reverse mode, passing reverse is fine

Reply:
So the output of guess activity only depends on whether the mode is forward or reverse, not on any other aspect of the mode? Is it possible to document this so that users can rely on that?

Reply:
Added activity to `Seed`